src/callback_requests.rs

```diff
@@ -357,3 +365,56 @@ impl From<kubewarden_policy_sdk::host_capabilities::kubernetes::GetResourceReque
         }
     }
 }
+
+impl From<SubjectAccessReview> for CallbackRequestType {
+    fn from(req: SubjectAccessReview) -> Self {
+        CallbackRequestType::KubernetesCanI {
+            subject_access_review: Box::new(SubjectAccessReviewWrapper(req)),
+        }
+    }
+}
+
+/// Wrapper type to allow users to pass a `SubjectAccessReview` to the
+/// `KubernetesCanI` request. The original types does not implement the Eq trait.
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct SubjectAccessReviewWrapper(pub SubjectAccessReview);
+
+impl Eq for SubjectAccessReviewWrapper {}
+
+/// Implementing the AsRef trait for SubjectAccessReviewWrapper to allow
+/// easy access to the inner SubjectAccessReview type.
+impl AsRef<SubjectAccessReview> for SubjectAccessReviewWrapper {
+    fn as_ref(&self) -> &SubjectAccessReview {
+        &self.0
+    }
+}
+
+/// Implementing the Display trait for SubjectAccessReviewWrapper to simplify the
+/// cache key generation. This implementation uses a hasher to generate a unique
+/// identifier for the SubjectAccessReviewWrapper instance.
+impl Display for SubjectAccessReviewWrapper {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut hasher = DefaultHasher::new();
+        self.hash(&mut hasher);
+        write!(f, "SubjectAccessReviewWrapper({})", hasher.finish())
+    }
+}
+
+/// Implementing the Hash trait for SubjectAccessReviewWrapper to be used in cache key
+impl Hash for SubjectAccessReviewWrapper {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.0.spec.user.hash(state);
+        if let Some(ref resource_attributes) = self.0.spec.resource_attributes {
+            resource_attributes.namespace.hash(state);
+            resource_attributes.verb.hash(state);
+            resource_attributes.resource.hash(state);
+        }
+        if let Some(ref non_resource_attributes) = self.0.spec.non_resource_attributes {
+            non_resource_attributes.path.hash(state);
+            non_resource_attributes.verb.hash(state);
+        }
+        self.0.spec.groups.hash(state);
```

Comment:
This is an array, maybe it would be good to order it before doing the hash, as different orders means different hashes therefore a different cache key.

Reply:
About this, I'm double guessing myself. I'm now scared of people creating a DOS attack by providing a very big groups array in the worst order possible, making the caller policy time out.

Reply:
On second thought, I prefer to not order the groups array. Just as other vectors in host capabilities aren't order (vectors of KeylessInfo, etc in the sigstore host capabilities).